provider/ec2/environ.go

```diff
@@ -480,14 +480,18 @@ func (e *environ) StartInstance(args environs.StartInstanceParams) (_ *environs.
 		)
 	}
 	logger.Debugf("ec2 user data; %d bytes", len(userData))
-	var apiPort int
+	apiPorts := make([]int, 0, 2)
 	if args.InstanceConfig.Controller != nil {
-		apiPort = args.InstanceConfig.Controller.Config.APIPort()
+		apiPorts = append(apiPorts, args.InstanceConfig.Controller.Config.APIPort())
+		if args.InstanceConfig.Controller.Config.AutocertDNSName() != "" {
+			// Open port 80 as well as it handles Let's Encrypt HTTP challenge.
+			apiPorts = append(apiPorts, 80)
```

Comment:
What if APIPort is already 80? Should this return an error in that case?

Reply:
AFAICT if there is an DNS name then the API port is always set to 443.